packages/base/room.gts

```diff
@@ -606,9 +615,12 @@ export class RoomField extends FieldDef {
         `Expected to find ${fragments[0].data.totalParts} fragments for fragment of event id ${eventId} but found ${fragments.length} fragments`,
       );
     }
-    return JSON.parse(
+    let cardDoc = JSON.parse(
       fragments.map((f) => f.data.cardFragment).join(''),
     ) as LooseSingleCardDocument;
+
+    cardHashes.set(md5(JSON.stringify(cardDoc)), eventId);
```

Comment:
We are JSON.parse-ing then, JSON.stringify-ing the same thing. Let's avoid the extra unneeded stringify:

```suggestion
    let cardSource = fragments.map((f) => f.data.cardFragment).join('');
    let cardDoc = JSON.parse(cardSource) as LooseSingleCardDocument;
    cardHashes.set(md5(cardSource), eventId);
```